### PR TITLE
chore: decoupled svg imports from main bundle

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Decoupled svg imports from main bundle, this is an 800kb file and should be lazily loaded when we actually need an svg. The previous implementation is incorrect.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11178699547>
> Commit: c6ee0c31abfe0251587c6a2ee522c755e812ca9e
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11178699547&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Fri, 04 Oct 2024 13:28:17 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `Icon` component to load SVG icons asynchronously, improving efficiency and responsiveness.
	- Introduced a caching mechanism to optimize SVG imports.
	- Added a new dependency, `webpack-bundle-analyzer`, to analyze bundle size and structure during the build process.

- **Bug Fixes**
	- Resolved issues related to redundant SVG imports through lazy loading.

- **Chores**
	- Removed optimization settings for code minimization in the development configuration, improving the development experience by reducing visual clutter.
	- Updated `devServer` configuration to suppress warnings and errors in the client overlay.

### CODE REVIEW

File: app/client/craco.common.config.js [hunk 1]
```diff
@@ -141,6 +142,7 @@ module.exports = {
             "./src/components/designSystems/blueprintjs/icon/index.js",
           ),
         ),
+        new BundleAnalyzerPlugin()
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/craco.common.config.js:145]: This should affect the speed of the dev build, right? Can we put this under the env flag? To run only when necessary.

File: app/client/craco.dev.config.js [hunk 2]
```diff
@@ -11,9 +11,6 @@ module.exports = merge(common, {
       },
     },
   },
-  optimization: {
```
Comment (KelvinOm) [inline app/client/craco.dev.config.js:14]: why?

**Review by KelvinOm — DISMISSED** [review]

**Review by rajatagrawal — APPROVED** [review]

**Pull request merged**